lib/open_weather/request.rb

```diff
@@ -34,7 +34,11 @@ def request(method, path, options)
         when :post, :put
           request.path = path
           request.params = { appid: options.delete(:appid) }
-          request.body = options.to_json unless options.empty?
+          if options.key?(:body)
+            request.body = options.delete(:body).to_json
+          elsif options.present?
```

Comment:
`options.present?` is always `true`, right above you check `options.key?` which implies that.

Reply:
Oops, also forgot this isn't Rails >_<